### ISSUE TITLE
UndesirableTypeInference false positive?

### ISSUE THREAD

**Issue by benblack86** [HairyFotr/linter#56]
For example, the below cause the error "Inferred type Map[String,Any]. (This might not be what you've intended)". Since the method is defined as returning Map[String, Any] I would think this is acceptable code.

```
val exifMap = parseExif()
def parseExif(): Map[String, Any] = {}
```
